core/deployment/src/main/java/io/quarkus/deployment/ResolvedJVMRequirements.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.deployment;
+
+import java.util.List;
+import java.util.jar.Attributes;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.deployment.builditem.ModuleOpenBuildItem;
+
+/**
+ * Represents requirements and restrictions on the runtime.
+ * Currently only used to track add-opens requirements; I'd like to eventually
+ * support tracking, for example, the required JVM version as we start to see
+ * some extensions having stronger opinions.
+ * Another use could be, for example, to force enabling experimental features such
+ * as needing jdk.incubator.vector (which would also inherently bump the minimum
+ * JVM version).
+ */
+public final class ResolvedJVMRequirements extends SimpleBuildItem {
+    private static final Logger LOG = Logger.getLogger(ResolvedJVMRequirements.class);
+    private static final Attributes.Name ADD_OPENS_JARATTRIBUTENAME = new Attributes.Name("Add-Opens");
+    private List<String> modulesToAddOpens;
+
+    public ResolvedJVMRequirements(List<ModuleOpenBuildItem> addOpens) {
+        this.modulesToAddOpens = addOpens.stream().map(ModuleOpenBuildItem::moduleName).distinct().sorted().toList();
```

Comment:
This should just bother build time right?
In any case we should think to unify our "programming style" saving expensive code pattern for code which run mostly once or just few times: it will help to keep built time under control too...
I know it's ugly, but it is what it is...